mindmeld/components/dialogue.py

```diff
@@ -689,6 +692,189 @@ def _get_dialogue_handler(self, dialogue_state):
         return handler
 
 
+class AutoEntityFilling():
+    """A special dialogue flow sublcass to implement Automatic Entitiy (Slot) Filling
+    (AEF) that allows developers to prompt users for completing the missing
+    requirements for entity slots.
+
+    Attributes:
+        app (Application): The application that initializes this flow.
+    """
+
+    logger = mod_logger.getChild("AutoEntityFilling")
+    """Class logger."""
+
+    def __init__(self, entrance_handler, entity_form, retry_attempts, app, **kwargs):
```

Comment:
We should set a default value to `retry_attempts` and validate `entity_form`, at least when `entity_form` is None we should throw an error

```suggestion
    def __init__(self, entrance_handler, entity_form, retry_attempts=1, app, **kwargs):
```

Reply:
I've set this default value in app.py, at the point of object initialization